src/screens/SignTransactionSheet.tsx

```diff
@@ -0,0 +1,2242 @@
+  }, [
+    handleConfirmTransaction,
+    handleSignMessage,
+    isBalanceEnough,
+    isMessageRequest,
+    isValidGas,
+  ]);
+
+  const onPressSend = useCallback(async () => {
+    if (isAuthorizing) return;
+    setIsAuthorizing(true);
+    try {
+      await onConfirm();
+      setIsAuthorizing(false);
+    } catch (error) {
+      setIsAuthorizing(false);
+    }
+  }, [isAuthorizing, onConfirm]);
+
+  const submitFn = useCallback(async () => {
+    if (!isBalanceEnough) {
+      navigate(Routes.ADD_CASH_SHEET);
+      return;
+    }
+    if (accountInfo.isHardwareWallet) {
+      navigate(Routes.HARDWARE_WALLET_TX_NAVIGATOR, { submit: onPressSend });
+    } else {
+      await onPressSend();
+    }
+  }, [accountInfo.isHardwareWallet, isBalanceEnough, navigate, onPressSend]);
+
+  const onPressCancel = useCallback(() => onCancel(), [onCancel]);
+
+  return (
+    <Inset bottom={{ custom: safeAreaInsetValues.bottom + 20 }}>
+      <Box height="full" justifyContent="flex-end" width="full">
+        <Box
+          as={Animated.View}
+          borderRadius={39}
+          paddingBottom="24px"
+          paddingHorizontal="20px"
+          paddingTop="32px"
+          style={{
+            backgroundColor: isDarkMode ? '#191A1C' : surfacePrimary,
+          }}
+        >
+          <Stack space="24px">
+            <Inset horizontal="12px" right={{ custom: 110 }}>
+              <Inline alignVertical="center" space="12px" wrap={false}>
+                <Box
+                  height={{ custom: 44 }}
+                  style={{
+                    backgroundColor: isDarkMode
+                      ? globalColors.white10
+                      : '#FBFCFD',
+                    borderRadius: 12,
+                    shadowColor: isDarkMode ? colors.trueBlack : colors.dark,
+                    shadowOffset: {
+                      width: 0,
+                      height: 18,
+                    },
+                    shadowOpacity: isDarkMode ? 1 : 0.12,
+                    shadowRadius: 27,
+                  }}
+                  width={{ custom: 44 }}
+                >
+                  <Image
+                    source={{
+                      uri: maybeSignUri(transactionDetails.imageUrl, {
+                        w: 100,
+                      }),
+                    }}
+                    style={{ borderRadius: 12, height: 44, width: 44 }}
+                  />
+                </Box>
+                <Stack space="12px">
+                  <Inline
+                    alignVertical="center"
+                    space={{ custom: 5 }}
+                    wrap={false}
+                  >
+                    <Text
+                      color="label"
+                      numberOfLines={1}
+                      size="20pt"
+                      weight="heavy"
+                    >
+                      {transactionDetails.dappName}
+                    </Text>
+                    {false && <VerifiedBadge />}
```

Comment:
🤔

Reply:
yeah we can't verify urls via WC consistently but leaving there cause there may be path forward soon